graphstorm-processing/pyproject.toml

```
@@ -12,22 +12,21 @@ authors = [
 python = "~3.9.12"
 pyspark = ">=3.3.0, < 3.5.0"
 pyarrow = "~13.0.0"
-spacy = "3.6.0"
 boto3 = "~1.28.1"
 joblib = "^1.3.1"
 pandas = "^1.3.5"
 psutil = "^5.9.5"
 sagemaker = "^2.83.0"
 scipy = "^1.10.1"
 transformers = "^4.37.1"
+torch = {url = "https://download.pytorch.org/whl/cpu/torch-2.1.2%2Bcpu-cp39-cp39-linux_x86_64.whl"}
```

Have we tested if this now works on Mac M1 laptops?

---

Let's go with the platform-safe:

```suggestion
torch = [
    { url = "https://download.pytorch.org/whl/cpu/torch-2.2.0%2Bcpu-cp39-cp39-linux_x86_64.whl", markers = "sys_platform == 'linux' and platform_machine != 'aarch64'"},
    { url = "https://download.pytorch.org/whl/cpu/torch-2.2.0%2Bcpu-cp39-cp39-linux_x86_64.whl", markers = "sys_platform == 'darwin' and platform_machine != 'arm64'"},
    { url = "https://download.pytorch.org/whl/cpu/torch-2.2.0-cp39-none-macosx_11_0_arm64.whl", markers = "sys_platform == 'darwin' and platform_machine == 'arm64'"},
    { url = "https://download.pytorch.org/whl/cpu/torch-2.2.0-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", markers = "sys_platform == 'linux' and platform_machine == 'aarch64'"},
]
```

If there are image size issues we should solve them in the Dockerfile, not the project's dependencies.